nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/processors/JMSConsumer.java

```diff
@@ -71,10 +71,15 @@ public Void doInJms(Session session) throws JMSException {
                  * delivery and restarts with the oldest unacknowledged message
                  */
                 session.recover();
+                boolean isPubSub = JMSConsumer.this.jmsTemplate.isPubSubDomain();
                 Destination destination = JMSConsumer.this.jmsTemplate.getDestinationResolver().resolveDestinationName(
-                        session, destinationName, JMSConsumer.this.jmsTemplate.isPubSubDomain());
-                MessageConsumer msgConsumer = session.createConsumer(destination, null,
-                        JMSConsumer.this.jmsTemplate.isPubSubDomain());
+                        session, destinationName, isPubSub);
+                MessageConsumer msgConsumer;
+                if (isPubSub && subscriberName != null) {
+                    msgConsumer = session.createSharedDurableConsumer((Topic)destination, subscriberName);
```

Comment:
Since `Shared` and `Durable` are different consumer characteristics, we might want to let user specify two different properties in order to support 4 consumer types:

- Unshared nondurable subscriptions
- Unshared durable subscriptions
- Shared nondurable subscriptions
- Shared durable subscriptions